telegram/_forumtopic.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+#
+# A library that provides a Python interface to the Telegram Bot API
+# Copyright (C) 2015-2022
+# Leandro Toledo de Souza <devs@python-telegram-bot.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""This module contains objects related to Telegram forum topics."""
+
+from typing import TYPE_CHECKING, Optional
+
+from telegram._telegramobject import TelegramObject
+from telegram._utils.types import JSONDict
+
+if TYPE_CHECKING:
+    from telegram import Bot
+
+
+class ForumTopic(TelegramObject):
+    """
+    This object represents the content of a service message about a new forum topic created in
+    the chat.
+
+    Objects of this class are comparable in terms of equality. Two objects of this class are
+    considered equal, if their :attr:`message_thread_id` and :attr:`name` are equal.
```

Comment:
Should I include icon_color as well to be consistent with ForumTopicCreated? On the one hand we need consistency, on the other hand it is crucial that no topic is created with the same name, even though the color is different.

That was the reason why I asked about `icon_color` beiung used in comparison in ForumTopicCreated.

Reply:
Right, but will it not create the situation when a topic can be created with the same name but different colors? We would introduce a more complex logic (yes, the objects are not equal, but these attributes (`name`) are equal and so no, you, user, can't create a topic). Then a question would arise why we have such a strange equality check that isn't that helpful.

But maybe my logic is flawed somewhere. Or maybe it's OK to introduce separate checks, and it's fine that equality of objects and equality of the most important attribute is not the same thing.

Reply:
So I wasn't sure of the history of how to implement this, so I looked it up: This [PR](https://github.com/python-telegram-bot/python-telegram-bot/pull/1724#issue-554374164) states what I had said earlier, i.e using all required args. But of course this may not be suitable in all cases, see e.g. the [original issue](https://github.com/python-telegram-bot/python-telegram-bot/pull/604#issuecomment-301236909) and then again in the [next version](https://github.com/python-telegram-bot/python-telegram-bot/pull/1724#issuecomment-657589186).

> why we have such a strange equality check that isn't that helpful

in some cases, yeah. I personally don't really use the object for comparison and just use the attributes instead. But here I would vote to add `icon_color` for consistency reasons

@Bibo-Joshi

Reply:
> I would vote to add icon_color for consistency reasons

+1 from  me.

IMO the point of providing a custom `__eq__` still is to provide a comparison logic that is more meaningful that object identity. If the user disagrees with that logic, the can use their very own comparison logic manually.

Reply:
Done